configurator/src/main/java/org/picmg/configurator/StateSetTabController.java

```diff
@@ -84,30 +64,72 @@ public class StateSetTabController implements Initializable {
 	@FXML private ImageView outputUnitsImage;
 	@FXML private ImageView plusAccuracyImage;
 
+	StateSetTabController.StateSetTableData workingData = new StateSetTableData();
+
+	public StateSetTableData getWorkingData() {
+		return workingData;
+	}
+
 	//TODO:change to use state sensor data class
 	//SensorTableData workingData = new SensorTableData();
 
 	//	public SensorTableData getSensorTableData() {
 	//		return workingData;
 	//	}
 	public class StateSetTableData{
+		SimpleStringProperty stateSetVendorName = new SimpleStringProperty();
+		SimpleStringProperty stateSetVendorIANA = new SimpleStringProperty();
+		SimpleStringProperty stateSetId = new SimpleStringProperty();
+		List<OEMStateValueRecord> oemStateValueRecords = new LinkedList<>();
+
+		public List<OEMStateValueRecord> getOemStateValueRecords() {
+			return oemStateValueRecords;
+		}
+
+		public void setOemStateValueRecords(List<OEMStateValueRecord> oemStateValueRecords) {
+			this.oemStateValueRecords = oemStateValueRecords;
+		}
 
+		public String getStateSetId() {
+			return stateSetId.get();
+		}
+
+		public void setStateSetId(String stateSetId) {
+			this.stateSetId.set(stateSetId);
+		}
+
+		public void setStateSetVendorIANA(String stateSetVendorIANA) {
+			this.stateSetVendorIANA.set(stateSetVendorIANA);
+		}
+
+		public void setStateSetVendorName(String stateSetVendorName) {
+			this.stateSetVendorName.set(stateSetVendorName);
+		}
 	}
 	public class ValueRecords{
 
 	}
 
 	@FXML
 	void onStateSetVendorNameAction(ActionEvent event) {
-		//TODO: finish
+		if (stateSetVendorNameTextField.getText().isBlank()) vendorNameImage.setVisible(true);
+		else vendorNameImage.setVisible(false);
+		workingData.setStateSetVendorName(stateSetVendorNameTextField.getText());;
+		saveChangesButton.setDisable(!isValid());
```

Comment:
I think for each of these actions, we also need to set saveAsChangesButton disabled on isValid

Reply:
done